submodules/telemetry/__init__.py

```diff
@@ -1,166 +1,129 @@
+from submodules import command_ingest
 
-# FIXME: Rename to telemetry_send_loop()
-def telemetry_send():
-    """
-    Thread method to burst the telemetry every so often
-    """
-    # TODO: check battery levels before sending
-    global telem_packet_buffer, event_packet_buffer
-    time.sleep(60)  # Don't send packets straight away
+# Command Ingest - ability to register commands
+from submodules.command_ingest import command
 
-    while core.get_state() == Mode.NORMAL:
-        # TODO if (adcs.can_TJ_be_seen() == True and len(telem_packet_buffer) + len(event_packet_buffer) > 0):
-        if len(telem_packet_buffer) + len(event_packet_buffer) > 0:
-            telemetry_send_once()
-        time.sleep(config['telemetry']['send_interval'])
+# Log and error classes
+from core import error, log
 
 
-# FIXME: Rename to telemetry_send()
-def telemetry_send_once():
-    """
-    Immediately send telemetry packets in both telemetry and event packet queues
-    """
-    global telem_packet_buffer, event_packet_buffer
-    beg_count = len(telem_packet_buffer) + len(event_packet_buffer)
-    send()
-    logger.debug("Sent " + str(beg_count - len(telem_packet_buffer) -
-                               len(event_packet_buffer)) + " telemetry packets")
+logger = logging.getLogger("TELEMETRY")
 
+general_queue = collections.deque()
+log_stack = collections.deque()
+err_stack = collections.deque()
 
-@command("telem_burst")
-def telemetry_burst_command():
-    """
-    Burst command; ignores isTJseen and other checks.
-    """
-    send(ignoreADCS=True)
-
+packet_lock = Lock()
 
-def enqueue_event_message(event):
+def enqueue(message) -> None:
     """
-    Enqueue an event message.
-    Event messages are a code representing the error, for instance "I01" for IMU error #1 which is to be looked up in a table upon receipt.
-    :param event: message to enqueue following code above. There is a table of codes in Google Sheets. Maximum three bytes.
+    Enqueue a message onto the general queue, to be processed later by thread decide()
+    :param message: The message to push onto general queue, log/error class, or command (string - must begin with semicolon, see command_ingest's readme)
+    :return: Nothing
     """
-
-    if len(event.encode('utf-8')) != 3:
-        logger.error("Event message larger than 3 bytes, message is " +
-                     str(len(event.encode('utf-8'))) + " bytes long")
+    if not ((type(message) is str and message[0] == ';') or type(message) is error.Error or type(message) is log.Log):
+        logger.error("Attempted to enqueue invalid message")
         return
+    general_queue.append(message)
 
-    global event_packet_buffer, packet_lock
-    with packet_lock:
-        packet = "Z"
-        packet += str(base64.b64encode(struct.pack('d', time.time())))
-        packet += str(base64.b64encode(event))
-        event_packet_buffer.append(packet)
-
-
-def enqueue_submodule_packet(packet):
+@command("telem_dump")
```

Comment:
we are removing the `command` decorator with manually recording each function in a db in `command_ingest`

Reply:
Removed all of them in 963480f